adapters/repos/db/vector/hnsw/index.go

```diff
@@ -214,6 +214,10 @@ func New(cfg Config, uc ent.UserConfig, tombstoneCallbacks, shardCompactionCallb
 		return nil, errors.Wrap(err, "invalid config")
 	}
 
+	if !asyncEnabled() && uc.PQ.Enabled {
```

Comment:
Unfortunately `New` is called both on initial creation and restore of the HNSW from disk. If you load, enable PQ, and reload (without async indexing enabled) you get:

```
ERRO[0001] Unable to load shard tV8ILAedPy1u: init shard "vector_tV8ILAedPy1u": hnsw index: impossible to create an hnsw instace with PQ enabled under sync indexing  error="init shard \"vector_tV8ILAedPy1u\": hnsw index: impossible to create an hnsw instace with PQ enabled under sync indexing"
```

This type of validation needs to ensure it only runs on class creation and not restore.